packages/orm/src/createOrUpdatePartial.ts

```diff
@@ -138,6 +140,12 @@ export async function createOrUpdatePartial<T extends Entity>(
     return em.createPartial(constructor, _opts);
   } else {
     const entity = await em.load(constructor, id);
+    // For o2m and m2m .set to work, they need to be loaded so that they know what to remove.
+    // Note that we also have the `delete: true` pattern for flagging not only "remove" but "delete",
+    // for a parent's mutation to control the lifecycle of a child entity (i.e. line items).
+    // Musing: Maybe this should happen implicitly, like if a LineItem.parent is set to null, that
+    // LineItem knows to just `em.delete` itself? Instead of relying on hints from GraphQL mutations.
+    await Promise.all(collectionsToLoad.map((fieldName) => (entity as any)[fieldName].load()));
```

Comment:
Fwiw m2m collections could be optimized, where we don't have to load 1000 m2m rows from the db, just to know a `parent.rows = [a,b]` should delete 998 of them, i.e. we could instead inside of `M2MCollection.set` just tell the UoW "delete my existing m2m rows with bulk sql before doing my inserts", but I wanted to leave that as a follow up optimization.